src/request.js

```diff
@@ -146,19 +145,47 @@ class Request {
     /**
      * Stores information about an error that occurred during processing of this request.
      *
+     * You should always use Error instances when throwing errors in JavaScript.
+     *
+     * Nevertheless, to improve the debugging experience when using third party libraries
+     * that may not always throw an Error instance, the function performs a type
+     * inspection of the passed argument and attempts to extract as much information
+     * as possible, since just throwing a bad type error makes any debugging rather difficult.
+     *
      * @param {Error|String} errorOrMessage Error object or error message to be stored in the request.
      */
     pushErrorMessage(errorOrMessage) {
-        if (!_.isString(errorOrMessage) && !(errorOrMessage instanceof Error)) {
-            throw new Error('Parameter errorOrMessage must be a String or an instance of Error');
+        let message;
+        const type = typeof errorOrMessage;
+        switch (type) {
+        case 'undefined':
+        case 'boolean':
+        case 'number':
+        case 'function':
+            message = `Received: "${errorOrMessage}" of type: "${type}" instead of a proper message.`; break;
+        case 'string':
+            message = errorOrMessage; break;
+        case 'symbol':
+            message = errorOrMessage.toString(); break;
+        case 'object':
+            if (!errorOrMessage) {
+                message = 'Received: "null" instead of a proper message.';
+            } else if (errorOrMessage.message) {
+                message = errorOrMessage.message; // eslint-disable-line prefer-destructuring
+            } else {
+                try {
+                    message = JSON.stringify(errorOrMessage, null, 2);
```

Comment:
You should check `toString()` that might be there in a case of object and many other types